crates/evm/src/tests/fork_tests.rs

```diff
@@ -0,0 +1,807 @@
+    assert!(receipts[0].receipt.success);
+
+    // the to address balance should be equal to contract balance
+    assert_eq!(die_to_acc.balance, U256::from(contract_balance));
+
+    let db_account = DbAccount::new(new_contract_address);
+
+    // the storage should not be empty
+    assert_eq!(
+        db_account.storage.get(&U256::from(0), &mut working_set),
+        Some(U256::from(123))
+    );
+
+    // Test if we managed to send money to ocntract
+    assert_eq!(contract_info.balance, U256::from(0));
+
+    let die_to_contract = evm
+        .accounts
+        .get(&die_to_address, &mut working_set)
+        .expect("die to address should exist");
+
+    // the to address balance should be equal to double contract balance now that two selfdestructs have been called
+    assert_eq!(die_to_contract.balance, U256::from(2 * contract_balance));
+}
+
+// tests second part (last one) of https://eips.ethereum.org/EIPS/eip-6780
+#[test]
+fn test_self_destructing_constructor() {
+    let contract_balance: u128 = 1000000000000000;
+
+    // address used in selfdestruct
+    let die_to_address = address!("11115497b157177315e1204f52e588b393111111");
+
+    let (config, dev_signer, contract_addr) =
+        get_evm_config(U256::from_str("100000000000000000000").unwrap(), None);
+
+    let (mut evm, mut working_set) = get_evm(&config);
+    let l1_fee_rate = 0;
+    let l2_height = 2;
+
+    let soft_confirmation_info = HookSoftConfirmationInfo {
+        l2_height,
+        da_slot_hash: [5u8; 32],
+        da_slot_height: 1,
+        da_slot_txs_commitment: [42u8; 32],
+        pre_state_root: [10u8; 32].to_vec(),
+        current_spec: SovSpecId::Fork1,
+        pub_key: vec![],
+        deposit_data: vec![],
+        l1_fee_rate,
+        timestamp: 0,
+    };
+    let contract = SelfdestructingConstructorContract::default();
+
+    let constructed_bytecode = contract.construct(die_to_address);
+
+    evm.begin_soft_confirmation_hook(&soft_confirmation_info, &mut working_set);
+    {
+        let sender_address = generate_address::<C>("sender");
+        let context = C::new(sender_address, l2_height, SovSpecId::Fork1, l1_fee_rate);
+
+        // deploy selfdestruct contract
+        let rlp_transactions = vec![create_contract_message_with_bytecode(
+            &dev_signer,
+            0,
+            constructed_bytecode,
+            Some(contract_balance),
+        )];
+
+        evm.call(
+            CallMessage {
+                txs: rlp_transactions,
+            },
+            &context,
+            &mut working_set,
+        )
+        .unwrap();
+    }
+    evm.end_soft_confirmation_hook(&soft_confirmation_info, &mut working_set);
+    evm.finalize_hook(&[99u8; 32].into(), &mut working_set.accessory_state());
+
+    let contract_info = evm.accounts.get(&contract_addr, &mut working_set);
+    // Contract should not exist as it is created and selfdestructed in the same transaction
+    assert!(contract_info.is_none());
```

Comment:
also check if code is none.

keep in mind since we do offchain code now you should check there. check code too though

Reply:
how can I check the code If I don't know the code hash tho
since code is a statemap I cannot iterate over it or check it's size either

Reply:
figured it out with `solc bin-runtime` option